altimeter/aws/resource/ec2/elastic_ip.py

```diff
@@ -0,0 +1,55 @@
+"""Resource for ElasticIPs"""
+from typing import Type
+
+from botocore.client import BaseClient
+
+from altimeter.aws.resource.resource_spec import ListFromAWSResult
+from altimeter.aws.resource.ec2 import EC2ResourceSpec
+from altimeter.core.graph.field.scalar_field import ScalarField
+from altimeter.core.graph.field.tags_field import TagsField
+from altimeter.core.graph.schema import Schema
+
+
+class ElasticIPResourceSpec(EC2ResourceSpec):
+    """Resource for ElasticIPs"""
+
+    type_name = "elastic-ip"
+    schema = Schema(
+        ScalarField("PublicIp"),
+        ScalarField("AllocationId"),
```

Comment:
It seems that `AllocationId` is optional. It is not used when in EC2-Classic.

In [moto](https://github.com/getmoto/moto/blob/a30dbafc93f9fa213a03c16c7378b55689434e5a/moto/ec2/models/elastic_ip_addresses.py#L32):

```python
        self.allocation_id = random_eip_allocation_id() if domain == "vpc" else None
```

From [AWS API docs](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Address.html):

> allocationId
>
> The ID representing the allocation of the address for use with EC2-VPC.

And it seems that [CloudFormation requires different attributes](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-eip-association.html) for `AWS::EC2::EIPAssociation` depending on using EC2-VPC and EC2-Classic:

> [EC2-VPC] You must specify AllocationId and either InstanceId, NetworkInterfaceId, or PrivateIpAddress.
>
> [EC2-Classic] You must specify EIP and InstanceId.

Reply:
I though they were because the documentation for the `describe_addresses` method of Boto3 was not clear. It's very good you manage to find more information about it, I'm changing the the parameter to optional.